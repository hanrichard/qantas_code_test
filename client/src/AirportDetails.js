import React from 'react';
import gql from 'graphql-tag';
import { useQuery } from '@apollo/react-hooks';
import { Link } from 'react-router-dom';

export const GET_AIRPORT_QUERY = gql`
    {
        airports {
            airportCode
            airportName
            location {
                latitude
                longitude
            }
            city {
                timeZoneName
            }
            country {
                countryName
            }
        }
    }
`;

const AirportDetails = props => {
    const { loading, error, data } = useQuery(GET_AIRPORT_QUERY);

    if (loading) return <div>loading...</div>;

    if (error) return <div>{error.message}</div>;

    const airport = data.airports.filter(airport => {
        return airport.airportCode === props.match.params.id;
    })[0];

    return (
        <div>
            <div>Airport Name: {airport.airportName}</div>

            <Link to="/">Back</Link>
        </div>
    );
};

export default AirportDetails;
