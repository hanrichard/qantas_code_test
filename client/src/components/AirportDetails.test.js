import React from 'react';
import { shallow, configure } from 'enzyme';
import AirportDetails from './AirportDetails';
import Adapter from 'enzyme-adapter-react-16';
import Button from '@material-ui/core/Button';
import Renderer from 'react-test-renderer';

configure({ adapter: new Adapter() });

function setup() {
    const props = {
        airport: {
            airportCode: 'AAA',
            airportName: 'Anaa',
            location: {
                latitude: '17.25',
                longitude: '145.3',
            },
            city: {
                timeZoneName: 'Pacific/Tahiti',
            },
            country: {
                countryName: 'French Polynesia',
            },
        },
    };

    const shallowWrapper = shallow(<AirportDetails {...props} />);

    return {
        props,
        shallowWrapper,
    };
}

describe('Airport', () => {
    const { shallowWrapper } = setup();
    it('should match snapshot', () => {
        const airport = Renderer.create(`<BrowserRouter>shallowWrapper</BrowserRouter>`).toJSON();
        expect(airport).toMatchSnapshot();
    });
});

describe('Airport details component', () => {
    const { shallowWrapper } = setup();
    it('should render self and its subcomponents', () => {
        expect(shallowWrapper.text().includes('Anaa')).toBe(true);
        expect(shallowWrapper.text().includes('17.25')).toBe(true);
        expect(shallowWrapper.text().includes('145.3')).toBe(true);
        expect(shallowWrapper.text().includes('Pacific/Tahiti')).toBe(true);
        expect(shallowWrapper.find(Button)).toHaveLength(1);
    });

    it('should render no results, when id is wrong', () => {
        shallowWrapper.setProps({ airport: null });
        expect(shallowWrapper.text().includes('145.3')).toBe(false);
        expect(shallowWrapper.text().includes('No result')).toBe(true);
    });
});
