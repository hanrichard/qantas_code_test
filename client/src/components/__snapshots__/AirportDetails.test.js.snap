// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Airport should match snapshot 1`] = `"<BrowserRouter>shallowWrapper</BrowserRouter>"`;
